internal/services/netapp/netapp_volume_group_resource.go

```diff
@@ -0,0 +1,642 @@
+	return "azurerm_netapp_volume_group"
+}
+
+func (r NetAppVolumeGroupResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return volumegroups.ValidateVolumeGroupID
+}
+
+func (r NetAppVolumeGroupResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"account_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: netAppValidate.AccountName,
+		},
+
+		"group_description": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"application_type": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(volumegroups.ApplicationTypeSAPNegativeHANA),
+			}, false),
+		},
+
+		"application_identifier": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringLenBetween(1, 3),
+		},
+
+		"volume": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MinItems: 2,
+			MaxItems: 5,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"id": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: netAppValidate.VolumeName,
+					},
+
+					"capacity_pool_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: azure.ValidateResourceID,
+					},
+
+					"proximity_placement_group_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: azure.ValidateResourceID,
+					},
+
+					"volume_spec_name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringInSlice(PossibleValuesForVolumeSpecName(), false),
+					},
+
+					"volume_path": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: netAppValidate.VolumePath,
+					},
+
+					"service_level": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ForceNew: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(volumegroups.ServiceLevelPremium),
+							string(volumegroups.ServiceLevelStandard),
+							string(volumegroups.ServiceLevelUltra),
+						}, false),
+					},
+
+					"subnet_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: azure.ValidateResourceID,
+					},
+
+					"protocols": {
+						Type:     pluginsdk.TypeList,
```

Comment:
We might have in the future support for dual protocol so I would not change this, otherwise the change it back to array will require a breaking change.

Reply:
That makes sense!